include/boost/url/impl/const_string.ipp

```diff
@@ -128,69 +128,70 @@ const_string::
 const_string(
     result const& r) noexcept
     : string_view(
-        r.data, r.size)
-    , p_(r.p)
+        r.data, r.size),
+        data_(r.p)
 {
 }
 
 const_string::
 ~const_string()
 {
-    p_->release(size());
+    if (!is_small(size())) {
+        data_.p_->release(size());
+    }
 }
 
 const_string::
 const_string() noexcept
-    : string_view()
-    , p_([]
-    {
-        struct empty_string final : base
-        {
-            void
-            destroy(std::size_t) noexcept override
-            {
-            }
-        };
-
-        struct deleter
-        {
-            empty_string* p;
-
-            ~deleter()
-            {
-                p->release(0);
-            }
-        };
-
-        static empty_string cs{};
-        static deleter d{&cs};
-
-        ++cs.refs;
-
-        return &cs;
-    }())
+    : string_view(data_.buf_, 0)
 {
 }
 
 const_string::
 const_string(
     const_string const& other) noexcept
-    : string_view(other)
-    , p_(other.p_)
+    : string_view()
```

Comment:
do we need this

Reply:
Removing this would lead to a warning in some compilers

```
warning: base class ‘class boost::core::basic_string_view<char>’ should be explicitly initialized in the copy constructor
```

Reply:
Same as another comment above

Reply:
That's a weird warning, I'm not sure its valid. I thought the base would be implicitly default constructed?